docs/source/basics/defining_environment.md

```diff
@@ -0,0 +1,239 @@
+```
+</details>
+<br />
+
+### Environment Variables
+Environment variables found in an `environment` file will be used to set environment variables inside the workflow environment.
+
+<details>
+<summary>Example File</summary>
+
+```
+BOWTIE2_INDEXES=reference
+PATH="$PATH:/root/bowtie2"
+```
+</details>
+<br />
+
+<details>
+<summary>Docker commands</summary>
+
+```Dockerfile
+env {line1}
+env {line2}
+...
+```
+</details>
+<br />
+
+---
+<br>
+
+To save the generated Dockerfile in the workflow directory, run `latch dockerfile {path_to_workflow_directory}`. The generated Dockefile will be used during subsequent `latch register` and `latch develop` commands to build the workflow environment, and it can be user modified. To go back to the autogenerated Dockerfile, delete the Dockerfile in the workflow directory.
+
+To understand the generated Dockefile, here is an example with instructive comments generated by running `latch init --template subprocess --dockerfile example_workflow`:
+
+```Dockerfile
+# latch base image + dependencies for latch SDK --- removing these will break the workflow
+from 812206152185.dkr.ecr.us-west-2.amazonaws.com/latch-base:ace9-main
+run pip install latch==2.12.1
+run mkdir /opt/latch
+
+# install system requirements
+copy system-requirements.txt /opt/latch/system-requirements.txt
+run apt-get update --yes && xargs apt-get install --yes </opt/latch/system-requirements.txt
+
+# copy all code from package (use .dockerignore to skip files)
+copy . /root/
+
+# set environment variables
+env BOWTIE2_INDEXES=reference
+
+# latch internal tagging system + expected root directory --- changing these lines will break the workflow
+arg tag
+env FLYTE_INTERNAL_IMAGE $tag
+workdir /root
+```
+
+Latch has three base images, one without hardware acceleration drivers, one with CUDA drivers, and one with OPENCL drivers. To use the CUDA or OPENCL base image, use the `--cuda` or `--opencl` flags when running `latch init`.
+
+## Ignoring files
+
+In the above Dockerfile, the line `copy . /root/` recursively copies all files and directories from the workflow directory into the Docker image. Even if files in the workflow directory are unused by the workflow, they will still be copied. Having unnecessary files in the image increases its size, slowing workflow execution and registration. Additionally, any changes to said files will cause image rebuilding when running `latch register` or `latch develop` on the workflow, making the process slower. To mitigate this, Latch uses a `.dockerignore` file in the workflow directory. This file should contain a list of files and directories that should be skipped when copying files into your container, as documented [here](https://docs.docker.com/engine/reference/builder/#dockerignore-file). Each workflow begins with a default `.dockerignore` file that ignores common pesky files (for example, `.latch_report.tar.gz` and other auto-generated files).
```

Comment:
- make independent from the example
- some rephrasing
- a few extra details
- dropped the part about `latch develop`/`latch register` cuz i dont actually think it adds anything to the previous sentence
- "pesky" is just funny
- mention "proportional to size" to explain that number of files doesn't really matter but large files matter a lot
```suggestion
By default, all files in the workflow root directory are included in the workflow build. Any unnecessary files will increase the resulting workflow container image size and increase registration and startup time proportional to their size.

To exclude files from the build use [a `.dockerignore`.](https://docs.docker.com/engine/reference/builder/#dockerignore-file) Files can be specified one at a time or using glob patterns.

The default `.dockerignore` includes files auto-generated by Latch.
```